Scripts/post.js

```diff
@@ -0,0 +1,298 @@
+    : "";
+
+  postContent.innerHTML = `
+  <div class="card-body">
+  ${authorHtml}
+  <h5 class="card-title">${post.title}</h5>
+  ${mediaHtml}
+  <p class="card-text mt-3">${post.body}</p>
+</div>
+<button class="btn btn-danger btn-sm delete-button" data-post-id="${post.id}">
+<i class="fa-solid fa-trash-can"></i> Delete
+</button>
+  `;
+  container.appendChild(postContent);
+
+  attachDeleteListener();
+  // Add Like and Dislike buttons
+  addLikeDislikeSection(container);
+  // Comment Section
+  createCommentSection(container, post.id);
+}
+
+// Function to add like and dislike section
+function addLikeDislikeSection(container) {
+  let likes = 0;
+  let dislikes = 0;
+
+  const likeDislikeSection = document.createElement("div");
+  likeDislikeSection.className = "like-dislike-buttons my-3";
+  likeDislikeSection.innerHTML = `
+      <button class="btn btn-success like-button">Like</button>
+      <button class="btn btn-danger dislike-button">Dislike</button>
+      <span class="likes">Likes: ${likes}</span>
+      <span class="dislikes">Dislikes: ${dislikes}</span>
+  `;
+  container.appendChild(likeDislikeSection);
+
+  const likeButton = likeDislikeSection.querySelector(".like-button");
+  const dislikeButton = likeDislikeSection.querySelector(".dislike-button");
+  const likesElement = likeDislikeSection.querySelector(".likes");
+  const dislikesElement = likeDislikeSection.querySelector(".dislikes");
+
+  likeButton.addEventListener("click", () => {
+    likes++;
+    updateLikesDislikes(likesElement, dislikesElement, likes, dislikes);
+  });
+
+  dislikeButton.addEventListener("click", () => {
+    dislikes++;
+    updateLikesDislikes(likesElement, dislikesElement, likes, dislikes);
+  });
+}
+
+// Function to update likes and dislikes counts
+function updateLikesDislikes(likesElement, dislikesElement, likes, dislikes) {
+  likesElement.textContent = `Likes: ${likes}`;
+  dislikesElement.textContent = `Dislikes: ${dislikes}`;
+}
+
+// Listener for the delete button
+function attachDeleteListener() {
+  const deleteButton = document.querySelector(".delete-button");
+  if (deleteButton) {
+    deleteButton.addEventListener("click", async (event) => {
+      const postId = event.target.dataset.postId;
+      try {
+        const deletedPostResponse = await deletePost(postId);
+        if (deletedPostResponse.success) {
+          removePostFromUI(postId);
+        } else {
+          console.error(
+            "Failed to delete post:",
+            deletedPostResponse.error || "Unknown error"
+          );
+        }
+      } catch (error) {
+        console.error("Error deleting post:", error);
+      }
+    });
+  }
+}
+
+async function deletePost(postId) {
+  const token = localStorage.getItem("jwtToken");
+  try {
+    const response = await fetch(`${API_POSTS_URL}/${postId}`, {
+      method: "DELETE",
+      headers: {
+        Authorization: `Bearer ${token}`,
+      },
+    });
+
+    if (response.status === 204) {
```

Comment:
Don't check for status === 204, just check if(response.ok)

Should solve the problem.